packages/assets-controllers/src/MultichainAssetsRatesController/MultichainAssetsRatesController.ts

```diff
@@ -315,6 +333,67 @@ export class MultichainAssetsRatesController extends StaticIntervalPollingContro
     });
   }
 
+  /**
+   * Fetches historical prices for the current account
+   *
+   * @param asset - The asset to fetch historical prices for.
+   * @returns The historical prices.
+   */
+  async fetchHistoricalPricesForAsset(asset: CaipAssetType): Promise<void> {
+    const releaseLock = await this.#mutex.acquire();
+    return (async () => {
+      if (!this.isActive) {
+        return;
+      }
+      const currentCaipCurrency =
+        MAP_CAIP_CURRENCIES[this.#currentCurrency] ?? MAP_CAIP_CURRENCIES.usd;
+      // Check if we already have historical prices for this asset and currency
+      const historicalPriceExpirationTime =
+        this.state.historicalPrices[asset]?.[this.#currentCurrency]
+          ?.expirationTime ?? null;
+
+      const historicalPriceHasExpired =
+        historicalPriceExpirationTime &&
+        historicalPriceExpirationTime < Date.now();
+
+      if (historicalPriceHasExpired === false) {
```

Comment:
Just want to double check if we want to capture other falsey values.
![Screenshot 2025-04-16 at 17 37 53](https://github.com/user-attachments/assets/9bb6824e-adc2-480c-b801-f05cffb80bc4)

Reply:
I think false is the only value we care about
      // 1- if expiration time exists and did not expire `historicalPriceExpirationTime` will be a number and `historicalPriceHasExpired` will be false => should do early return(check against false value)
      // 2- if expiration time exists and did expire `historicalPriceExpirationTime` will be a number and `historicalPriceHasExpired` will be true => should fetch new historical prices (check against false is valid)
      // 3- if expiration time does not exist,  `historicalPriceExpirationTime` will be undefined => `historicalPriceHasExpired` will be undefined => should fetch new historical prices (check against false is valid)